packages/components/src/store/slices/table/index.ts

```diff
@@ -0,0 +1,84 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Position } from '@models/Position';
+import { PlayingCard } from '@models/PlayingCard';
+import { Dimensions } from '@models/Dimensions';
+import { generateRandomCard } from '@utils/generate-card';
+import { generateRandomPositionWithinBoundaries } from '@utils/position-utils';
+import { generateRandomCardDeck } from '@utils/generate-card-deck';
+import { shuffleArray } from '@utils/array-utils';
+
+export interface CardState {
+  card?: PlayingCard;
+  cards?: PlayingCard[];
```

Comment:
To avoid having `card` and `cards` wouldn't be better to have something like:

```
data: PlayingCard | PlayingCard[]
```

Reply:
Yeah I couldn't find a satisfying name that would include both card and cards 😅 
`data` is okay I guess but it's a bit too generic :/ Maybe I could just name it `cards` and check if it is an array or not?

Reply:
Yeah let's go for it then. Just call it `cards`

Reply:
Moreover, shouldn't that be `Card | Deck` ?

Reply:
Actually no, if I use `Deck` as a type then I need to create a type guard to check if cards is a Deck or just a PlayingCard.
By using `PlayingCard | PlayingCard[]`, my check is just using `Array.isArray(cardState.cards)`. I feel like it is simpler